go/libraries/doltcore/env/environment.go

```diff
@@ -499,7 +499,7 @@ func (dEnv *DoltEnv) createDirectories(dir string) (string, error) {
 		}
 
 		if len(entries) == 1 {
-			if !entries[0].IsDir() || entries[0].Name() != TmpDirName {
+			if (!entries[0].IsDir() && entries[0].Name() != TmpDirName) && (!entries[0].IsDir() && entries[0].Name() != configFile) {
```

Comment:
This logic doesn't seem correct to me 🤔  Looking at the original logic... seems like we were originally allowing only a directory named `TmpDirName`, but this new logic wouldn't allow that anymore. 

I think you actually want something like this: 

```suggestion
entry := entries[0]
if (entry.IsDir && entry.Name() != TmpDirName) || (!entry.IsDir() && entry.Name() != configFile) {
```